internal/pkg/utils/time_test.go

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (C) 2023 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckMinInterval(t *testing.T) {
+	lc := logger.NewMockClient()
```

Comment:
Use mocks.LoggerClient{} and mock.On("Warnf"...) and mock.AssertCalled/AssertNotCalled to test if warning was logged appropriately or not.

Reply:
This suggestion is helpful to verify the logging client method being called, thanks.